src/presets/nodeless.ts

```diff
@@ -89,7 +89,20 @@ const nodeless: Preset & { alias: Map<string, string> } = {
     global: "unenv/runtime/node/_global",
     process: "unenv/runtime/node/process",
     Buffer: ["buffer", "Buffer"],
-    performance: "unenv/runtime/polyfill/performance",
+    performance: ["perf_hooks", "performance"],
+
+    // TODO: for some reason the following Performance* injectables are not honored
+    //       figure out why... is this some kind of ESBuild bug?
```

Comment:
@pi0 this one really puzzles me. I have no idea what's wrong here. In my testing `globalThis.performance` works as expected but `globalThis.Performance` (capital P) is `undefined`.

Even more oddly, if I replace line #92 with `performance: ["perf_hooks", "Performance"]`, my ESBuild setup replaces `globalThis.performance` with `node:perf_hooks#Performance` as instructed.

So it's as if something about the left-hand-side `Performance` caused the injection to not work. Very bizarre.

Any ideas what could be causing this?

Reply:
This is the only thing that remains unresolved in this PR as far as I know and we should probably get to the bottom of it before merging, because otherwise we regress on `globalThis.Performance*` APIs compared to `main` branch.

Reply:
Nitro uses rollup plugin for injection not sure if it might be a direct issue with unenv...

Reply:
Also we have https://github.com/IgorMinar/unenv/blob/4104cad8e86a33851ed2f6d18ae5f27dbedb726f/src/presets/nodeless.ts#L111 - other than explicitly disabled state of cloudflare for nitro i would always globally polyfill. Bundlers can always miss this kind of injections (even if today work, i seen it many times)

Reply:
We currently don't use any of the `polyfills` in our wrangler integration and rely solely on `injection`. I'd rather risk the bundler missing some call sites than always bringing in these polyfills and bloating the bundle size.

Unlike the usual usage, [our injection implementation in Wrangler](https://github.com/cloudflare/workers-sdk/blob/874da0e79de18d3661963f8d617209a1bd7816bf/packages/wrangler/src/deployment-bundle/esbuild-plugins/hybrid-nodejs-compat.ts#L114-L188) is side-effecty, so it takes only one successful injection for the global to be polyfilled. That way if a bundler misses a call site, we can have users write noop code in their app that is not missed and its side-effect will bring in the polyfill for all call sites. I think this tree-shakable design is a better trade off between accuracy and size.

Reply:
In any case, I'll play with this a bit more today to see if I can figure out what's going on.

Reply:
(thinking would it make sense unenv have an explicit `wrangler` preset?)

Reply:
I'm not sure. I don't think it's strictly necessary. The current cloudflare preset should work with wrangler and outside of wrangler. The only special thing about the preset is that `inject` and `polyfills` fields are redundant and you can chose to use one or the other or both.

If you want the tree-shakable design that wrangler uses, then you'll need to duplicate some of the code from wrangler and adjust it to whatever bundler you are using (if it's not esbuild already). This is however optional and not a requirement for using the cloudflare preset.

Reply:
Was thinking about next steps (preset test matrix), and mainly if it will give a safer escape hatch for unseen differences in the future (that nitro and wrangler need different behavior). But only a suggestion.

Reply:
I got to the bottom of the original issue. There was an issue in the code in wrangler that I fixed in https://github.com/cloudflare/workers-sdk/pull/6025. So I think this PR is good to go now, I'll remove this TODO.

Reply:
> Was thinking about next steps (preset test matrix we need to test cloudflare twice now with and without polyfill - inject only), and mainly if it will give a safer escape hatch for unseen differences in the future (that nitro and wrangler need different behavior). But only a suggestion.

I think that rather than creating a separate preset for wrangler, we should more explicitly document how to integrate unenv with various bundlers. It took me quite a while to figure out how to take the unenv preset and use it with ESBuild which wrangler uses, and the integration is not trivial at all. I can imagine a future where the [ESBuild plugin embedded in wrangler](https://github.com/cloudflare/workers-sdk/blob/main/packages/wrangler/src/deployment-bundle/esbuild-plugins/hybrid-nodejs-compat.ts) is a standalone npm package that anyone can use to take an unenv preset and use it with esbuild.

Reply:
Yeah same for rolluo integration in Nitro which could be externalized.

Reply:
Moved to https://github.com/unjs/unenv/issues/262, https://github.com/unjs/unenv/issues/263 for furthor discussions.